test.py

```diff
@@ -1,124 +1,527 @@
+        pl_handler = GithubPayloadHandler(None)
+        pl_handler.new_pr = Mock()
+        pl_handler.update_pr = Mock()
+        pl_handler.new_comment = Mock()
+
+        payload = {"action":"created", "issue":{"number":1}}
+        pl_handler.handle_payload(payload)
+        pl_handler.new_comment.assert_called_with("1", payload)
+
+        payload = {"action":"opened", "number":"1"}
+        pl_handler.handle_payload(payload)
+        pl_handler.new_pr.assert_called_with("1", payload)
+
+        payload = {"action":"synchronize", "number":"1"}
+        pl_handler.handle_payload(payload)
+        pl_handler.new_pr.update_pr("1", payload)
+
+        # should not break
+        payload = {"action":"other", "number":"1"}
+        pl_handler.handle_payload(payload)
+
+
+    def test_manage_pr_state(self):
+        self.github.get_labels = Mock(return_value=["S-awaiting-merge", "S-tests-failed", "S-needs-code-changes", "S-needs-rebase"])
+        pl_handler = GithubPayloadHandler(self.github)
+        full_cov_payload = {"action":"synchronize", "pull_request":{"mergeable":True}}
+        issue_num = 1
+        
+        pl_handler.manage_pr_state(issue_num, full_cov_payload)
+        self.github.get_labels.assert_called_with(issue_num)
+        calls = [ 
+            call("S-awaiting-merge", issue_num),
+            call("S-tests-failed", issue_num),
+            call("S-needs-code-changes", issue_num),
+            call("S-needs-rebase", issue_num)
+        ]
+        self.github.remove_label.assert_has_calls(calls)
+        self.github.add_label.assert_called_with("S-awaiting-review", issue_num)
+
+    # Not sure it's worth the effort to cover all branches of manage_pr_state
+
+
+    def test_new_comment_no_action(self):
+        pl_handler = GithubPayloadHandler(self.github)
+        pl_handler._find_reviewer = Mock()
+        payload = {"issue":{"state":"closed"}}
+        issue_num = 1
+
+        pl_handler.new_comment(issue_num, payload)
+        pl_handler._find_reviewer.assert_not_called()
+
+        payload = {"issue":{"state":"open","pull_request":1},"comment":{"user":{"login":self.github_user}}}
+        pl_handler.new_comment(issue_num, payload)
+        pl_handler._find_reviewer.assert_not_called()
+
+
+    def test_new_comment_approved(self):
+        self.github.set_assignee = Mock()
+        self.github.get_labels = Mock(return_value=["S-awaiting-review"])
+        pl_handler = GithubPayloadHandler(self.github)
+        payload = {"issue":{"state":"open","pull_request":1},"comment":{"user":{"login":"bors-servo"},"body":"\br?:@jdm Testing commit"}}
+        issue_num = 1
+
+        pl_handler.new_comment(issue_num, payload)
+        self.github.set_assignee.assert_called_with("jdm", issue_num)
+        self.github.get_labels.assert_called_with(issue_num)
+        self.github.remove_label.assert_called_with("S-awaiting-review", issue_num)
+        self.github.add_label.assert_called_with("S-awaiting-merge")
```

Comment:
This test is generally ok, but I don't see the value of asserting that get_labels was called. I'm always more interested in verifying that externally-visible rather than internal state changes occured.